ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala

```diff
@@ -164,42 +141,59 @@ final class EmberClientBuilder[F[_]: Concurrent: Timer: ContextShift] private (
           .withOnReaperException(_ => Applicative[F].unit)
       pool <- builder.build
     } yield {
-      val client = Client[F](request =>
+      val client = Client[F] { request =>
+        def getValidManaged: Resource[F, Managed[F, (RequestKeySocket[F], F[Unit])]] =
+          pool.take(RequestKey.fromRequest(request)).flatMap { managed =>
+            Resource
+              .liftF(managed.value._1.socket.isOpen)
+              .ifM(
+                managed.pure[Resource[F, *]],
+                // Already Closed,
+                // The Resource Scopes Aren't doing us anything
+                // if we have max removed from pool we will need to revisit
+                Resource.liftF(managed.canBeReused.set(Reusable.DontReuse)) >>
+                  getValidManaged
```

Comment:
This risks deadlocking the client, I think?  `managed` is still in scope when we loop and until we're done with whatever new one we get.  This has killed us on the redirect and retry middleware before.  In a loop, you need to release the old before allocating the new.  `Hotswap` gives an elegant CE3 implementation, but CE2 is much trickier.

Reply:
Okay, we should only be cycling through connections that are already allocated, unless `pool.take` starts returning a series of instantly closed ones.  We could put a safety valve here, but it's not the Retry bug redux I initially feared.